Signal/src/Profiles/OWSProfileManager.m

```diff
@@ -236,6 +242,11 @@ - (void)saveUserProfile:(UserProfile *)userProfile
 
         dispatch_async(dispatch_get_main_queue(), ^{
             if (isLocalUserProfile) {
+                [MultiDeviceProfileKeyUpdateJob runWithProfileKey:userProfile.profileKey
```

Comment:
Hmm. `saveUserProfile` isn't called by `localUserProfile` when `localUserProfile` generates the local profile for the first time.  

Nor should it, or this line above would become unsafe:

```
 BOOL isLocalUserProfile = userProfile == self.localUserProfile;
```
So for existing users who upgrade to a "post-profile world" for the first time, their profile key _won't_ be sync to desktop when it is generated.  

I guess this is okay now since only your primary can edit your profile but eventually won't we want to let desktop edit your profile?  It seems to me like we should sync this sooner if we can.

Reply:
We sync the profile key as soon as the user submits profile data. The value of distributing a profile key not yet associated with any profile data seems pretty negligible.

Reply:
If desktop wants to ship a profile editing, it won't be able to if primary doesn't share the profile key until it has edited the profile.  e.g. "Desktop can't edit profile until you've edited profile on primary".  

I'm trying to think about the simplest thing we could do.  It might be to replace this:

```
            if (_localUserProfile == nil) {
                DDLogInfo(@"%@ Building local profile.", self.tag);
                _localUserProfile = [[UserProfile alloc] initWithRecipientId:kLocalProfileUniqueId];
                _localUserProfile.profileKey = [OWSAES256Key generateRandomKey];
                [self.dbConnection readWriteWithBlock:^(YapDatabaseReadWriteTransaction *_Nonnull transaction) {
                    [_localUserProfile saveWithTransaction:transaction];
                }];
            }
```

with this:
```
            if (_localUserProfile == nil) {
                DDLogInfo(@"%@ Building local profile.", self.tag);
                _localUserProfile = [[UserProfile alloc] initWithRecipientId:kLocalProfileUniqueId];
                _localUserProfile.profileKey = [OWSAES256Key generateRandomKey];
                [self.dbConnection readWriteWithBlock:^(YapDatabaseReadWriteTransaction *_Nonnull transaction) {
                    [_localUserProfile saveWithTransaction:transaction];
                }];
                
                dispatch_async(dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_DEFAULT, 0), ^{
                    @synchronized(self)
                    {
                        [self saveUserProfile:_localUserProfile];
                    }
                });
            }
```

Reply:
> If desktop wants to ship a profile editing, it won't be able to if primary doesn't share the profile key until it has edited the profile. e.g. "Desktop can't edit profile until you've edited profile on primary".

Ok - in case that's part of the initial release, I'll make a PR with this change.